include/glow/Backends/Backend.h

```diff
@@ -88,23 +88,26 @@ class Backend {
   virtual bool shouldShareBuffers() const { return true; }
 
   /// Optimize the Function \p F given compilation mode \p mode.
-  void optimizeFunction(CompilationMode mode, Function *F);
+  void optimizeFunction(CompilationMode mode, Function *F) const;
 
   /// \returns true if Backend generated Instruction for Node \p N,
   /// using IRGenVisitor \p irgen.
   virtual bool generateInst(Node *N, IRGenVisitor &irgen) const {
     return false;
   }
 
+  virtual size_t getTraceEventDataSize() const { return 0; }
```

Comment:
If it's pure virtual it will break compile of backends that haven't implemented Trace Events, which isn't what we want (yet?). The default implementation essentially disables tracing which I think is a sensible default.

Reply:
Sounds good